src/sphinx_substitution_extensions/__init__.py

```diff
@@ -27,15 +30,16 @@
 # See https://github.com/PyCQA/doc8/pull/34.
 _SUBSTITUTION_OPTION_NAME = 'substitutions'
 
-if 'prompt' not in _EXISTING_DIRECTIVES:
-    MESSAGE = (
-        'sphinx-prompt must be in the conf.py extensions list before '
-        'sphinx_substitution_extensions'
-    )
-    LOGGER.error(MESSAGE)
-
-_EXISTING_PROMPT_DIRECTIVE: Directive = _EXISTING_DIRECTIVES['prompt']
-
+_EXISTING_PROMPT_DIRECTIVE = Directive
+if exists_dependency('sphinx-prompt'):
+    if 'prompt' not in _EXISTING_DIRECTIVES:
+        MESSAGE = (
+            'sphinx-prompt must be in the conf.py extensions list before '
+            'sphinx_substitution_extensions'
+        )
+        LOGGER.error(MESSAGE)
```

Comment:
I think that this is somewhat strange behaviour:

* I create a project which is not using `sphinx-substitution-extensions` and not `sphinx-prompt`.
* I `pip install sphinx-prompt`.
* I now get an error in my logs.

Is that right? If so, it might be confusing, for example, if a user has multiple Sphinx projects within one Python project.

What do you think?

Maybe this should become a `LOGGER.warn`?

Reply:
That's right, since you have the extension installed but you haven't added it yet to ``conf.py``.

I like the idea of changing it to a warning in case they have the dependency installed in case they share the same env for multiple Sphinx projects.